### PR TITLE
Rebase Propeller Engine on top of dev.v2

### PR DESCRIPTION
**Pull request by swimalick**
How I did this:
* rebased `fis-v2` on top of `wormat-v2-rebase`, removed most of the commits with `wormat:` in the front
* solved merge conflicts
* Changed the code from here https://github.com/swim-io/wormhole/pull/2/files#diff-fabd1e01edd6e3d9bcf318110b23c7b0540e9251406eb13fffe9781fa50a71aa to fit the new modular relayer interface

I commented on the files that I had to modify in order to fit the modular_relayer. Notably, `parseAndValidateVaa()` from `validation.ts` has been extracted into `backends/swim/listener.ts`

### CODE REVIEW

File: relayer/spy_relayer/src/backends/swim/listener.ts [hunk 1]
```diff
@@ -0,0 +1,288 @@
+/** The swim backend is relaying payload 3 with swim payload token bridge messages only */
```
Comment (swimalick) [inline relayer/spy_relayer/src/backends/swim/listener.ts:1]: This is one of the files that changed

File: relayer/spy_relayer/src/backends/swim/relayer.ts [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+  parseSwimPayload,
+  parseTransferWithArbPayload
+} from "../../utils/swim";
+
+function getChainConfigInfo(chainId: ChainId) {
+  const env = getRelayerEnvironment();
+  return env.supportedChains.find((x) => x.chainId === chainId);
+}
+
+/** Relayer for payload 3 token bridge messages with Swim Payload only */
```
Comment (swimalick) [inline relayer/spy_relayer/src/backends/swim/relayer.ts:41]: This is another file that changed

**Comment by wormat** [comment]
@swimalick I changed the base to `wormat-v2-rebased`. When we're confident this works I'll replace `wormat-v2` with the code from `wormat-v2-rebased`, and I guess you can do the same with `fis-v2`/`fis-v2-rebased`.

**Comment by wormat** [comment]
Alternatively you could just call this `alick-v2`.

**Review by wormat — COMMENTED** [review]
Haven't reviewed the spy changes but the rebase on top of the wormat/fis work looks good.

**Comment by swimalick** [comment]
We've succesfully gotten goerli -> bsc and vice versa working using this branch, so safe to say that it's working. 

I think going forward, it makes sense to never merge to dev.v2, and rather keep that as clean as possible so we can keep pulling wormhole changes into it. We should make a new branch for swim changes (lets call it `swim-v2` or even use `main`). That way, if wormhole pushes an important fix, we can pull it into our fork and simply rebase `swim-v2` on top of `dev.v2`

**Comment by wormat** [comment]
@swimalick OK to delete the branch?

**Comment by swimalick** [comment]
Yup! Done

**Pull request merged**